pyproject.toml

```diff
@@ -51,6 +52,10 @@ multi_line_output = 3
 testpaths = ["tests"]
 filterwarnings = [
     "error",
+    # Work around https://github.com/pytest-dev/pytest/issues/10977 for Python 3.12
+    'ignore:(ast\.Str|ast\.NameConstant|ast\.Num|Attribute s) is deprecated and will be removed.*:DeprecationWarning:',
```

Comment:
I'd rather *not* do this sort of masking for a pre-release; if we get closer to 3.12 final and it hasn't been fixed, then sure - but our 3.12 support is mostly for "canary" testing at this point (to catch issues like the tarball filter argument). I don't think we need to introduce workarounds for problems that *should* be fixed before 3.12 final.

Reply:
That's makes sense; I was kinda leery of it since there isn't a good way to track when it can come back out.

For at least the `pytest` error, I'm expecting them to fix it pretty quickly and have a new release available...not sure about the `dateutil` one.

I mostly just want GitHub to add proper support for `allow-failure` so CI doesn't look like it failed because testing failed on a non-critical platform...:(

Reply:
Furthermore, though, these changes won't get full coverage until the pytest `DeprecationWarning` is fixed.